content/articles/angular12-material-table/index.md

```diff
@@ -0,0 +1,301 @@
+### Introduction
+In this tutorial, I'll show you how to create Angular material tables and populate this table with some data. We'll then proceed and apply several functionalities to this table such as paging, filtering, and sorting.  
+
+### Table of contents
+
+- [Prerequisites](#prerequisites)
+- [ObjectivesAngular 12 Material Table With Pagination, Filtering, Sorting](#objectivesangular-12-material-table-with-pagination-filtering-sorting)
+- [Getting started with Angular Material](#getting-started-with-angular-material)
+- [Using Material Table to display data](#using-material-table-to-display-data)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+To follow along with this tutorial, you are required to have the following:
+
+- Basic knowledge of Angular. In this tutorial we'll be using Angular 12.
+- Knowledge on using Angular Material. It will help you get started quickly.
+- A complete Angular project installed on your machine.
+
+### Objectives
+This tutorial will teach you everything you need to use Angular Material tables. We'll build a complete project for implementing the Angular Material data table with multiple functionalities.
+
+### Getting started with Angular Material
+In this section, I'll show you how to set up your Angular project to start using Angular Material.
+
+In your project root, open the terminal and run the following commands:
+
+```bash
+ng add @angular/material
+```
+
+This command will prompt for a `yes`/`no` question as shown below:
+
+```bash
+.......
+# installation starts
+✔ Found compatible package version: @angular/material@12.1.3.
+..................
+Would you like to proceed? (Y/n) 
+```
+
+Enter `y` to proceed with the Angular Material installation. This will install all packages required for Material.
+
+![Material Installation](/engineering-education/angular-material-table/material-output.png)
+
+Now that we've successfully installed Material, let's now import material modules into our project.
+
+```js
+//import angular material modules here for our table
```

Comment:
There are many modules in Angular. Explain where the code here is being put. Are we creating a new module? Are we putting it inside the `app.module.ts`? Please this information. Provide the filenames for each codeblock.

Please do not leave gaps in your article. Do not let the reader refer to your code to understand what you mean.